### PR TITLE
bindings: prevent double free

### LINKED ISSUE #252: lxcfs 3.0.2 crashes on centos 7
Just tried lxcfs 3.0.2 on CentOS 7 with lxc 3.0.2 and within minutes lxcfs crashed with this

```
Aug 17 14:00:09 rh-vz01 lxcfs: *** Error in `/usr/bin/lxcfs': double free or corruption (out): 0x00007fed4c021cb0 ***
Aug 17 14:00:09 rh-vz01 lxcfs: ======= Backtrace: =========
Aug 17 14:00:09 rh-vz01 lxcfs: /lib64/libc.so.6(+0x81499)[0x7fed7d8b5499]
Aug 17 14:00:09 rh-vz01 lxcfs: /usr/lib64/lxcfs/liblxcfs.so(+0x51b8)[0x7fed7d6201b8]
Aug 17 14:00:09 rh-vz01 lxcfs: /usr/lib64/lxcfs/liblxcfs.so(cg_release+0xd)[0x7fed7d62403d]
Aug 17 14:00:09 rh-vz01 lxcfs: /usr/bin/lxcfs[0x401add]
Aug 17 14:00:09 rh-vz01 lxcfs: /lib64/libfuse.so.2(+0xcd12)[0x7fed7e02dd12]
Aug 17 14:00:09 rh-vz01 lxcfs: /lib64/libfuse.so.2(+0xf603)[0x7fed7e030603]
Aug 17 14:00:09 rh-vz01 lxcfs: /lib64/libfuse.so.2(+0x15fa4)[0x7fed7e036fa4]
Aug 17 14:00:09 rh-vz01 lxcfs: /lib64/libfuse.so.2(+0x16bcb)[0x7fed7e037bcb]
Aug 17 14:00:09 rh-vz01 lxcfs: /lib64/libfuse.so.2(+0x13461)[0x7fed7e034461]
Aug 17 14:00:09 rh-vz01 lxcfs: /lib64/libpthread.so.0(+0x7e25)[0x7fed7dc08e25]
Aug 17 14:00:09 rh-vz01 lxcfs: /lib64/libc.so.6(clone+0x6d)[0x7fed7d932bad]
Aug 17 14:00:09 rh-vz01 lxcfs: ======= Memory map: ========
Aug 17 14:00:09 rh-vz01 lxcfs: 00400000-00404000 r-xp 00000000 08:02 275404                             /usr/bin/lxcfs
Aug 17 14:00:09 rh-vz01 lxcfs: 00603000-00604000 r--p 00003000 08:02 275404                             /usr/bin/lxcfs
Aug 17 14:00:09 rh-vz01 lxcfs: 00604000-00605000 rw-p 00004000 08:02 275404                             /usr/bin/lxcfs
Aug 17 14:00:09 rh-vz01 lxcfs: 02114000-02156000 rw-p 00000000 00:00 0                                  [heap]
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed44000000-7fed4406d000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed4406d000-7fed48000000 ---p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed48000000-7fed48043000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed48043000-7fed4c000000 ---p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed4c000000-7fed4c085000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed4c085000-7fed50000000 ---p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed51ffc000-7fed51ffd000 ---p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed51ffd000-7fed527fd000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed527fd000-7fed527fe000 ---p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed527fe000-7fed52ffe000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed52ffe000-7fed52fff000 ---p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed52fff000-7fed537ff000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed537ff000-7fed53800000 ---p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed53800000-7fed54000000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed54000000-7fed54054000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed54054000-7fed58000000 ---p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed58000000-7fed58066000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed58066000-7fed5c000000 ---p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed5c000000-7fed5c064000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed5c064000-7fed60000000 ---p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed60000000-7fed6008e000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed6008e000-7fed64000000 ---p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed64000000-7fed64054000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed64054000-7fed68000000 ---p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed68000000-7fed68054000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed68054000-7fed6c000000 ---p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed6c000000-7fed6c086000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed6c086000-7fed70000000 ---p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed70000000-7fed7004c000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7004c000-7fed74000000 ---p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed747f9000-7fed747fa000 ---p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed747fa000-7fed74ffa000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 systemd: lxcfs.service: main process exited, code=killed, status=6/ABRT
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed74ffa000-7fed74ffb000 ---p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed74ffb000-7fed757fb000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed757fb000-7fed757fc000 ---p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed757fc000-7fed75ffc000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed75ffc000-7fed75ffd000 ---p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed75ffd000-7fed767fd000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed767fd000-7fed767fe000 ---p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed767fe000-7fed76ffe000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed76ffe000-7fed76fff000 ---p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed76fff000-7fed777ff000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed777ff000-7fed77800000 ---p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed77800000-7fed78000000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed78000000-7fed78054000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed78054000-7fed7c000000 ---p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7c403000-7fed7c418000 r-xp 00000000 08:02 270888                     /usr/lib64/libgcc_s-4.8.5-20150702.so.1
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7c418000-7fed7c617000 ---p 00015000 08:02 270888                     /usr/lib64/libgcc_s-4.8.5-20150702.so.1
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7c617000-7fed7c618000 r--p 00014000 08:02 270888                     /usr/lib64/libgcc_s-4.8.5-20150702.so.1
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7c618000-7fed7c619000 rw-p 00015000 08:02 270888                     /usr/lib64/libgcc_s-4.8.5-20150702.so.1
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7c619000-7fed7c61a000 ---p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7c61a000-7fed7ce1a000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7ce1a000-7fed7ce1b000 ---p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7ce1b000-7fed7d61b000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7d61b000-7fed7d62b000 r-xp 00000000 08:02 275406                     /usr/lib64/lxcfs/liblxcfs.so
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7d62b000-7fed7d82a000 ---p 00010000 08:02 275406                     /usr/lib64/lxcfs/liblxcfs.so
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7d82a000-7fed7d82b000 r--p 0000f000 08:02 275406                     /usr/lib64/lxcfs/liblxcfs.so
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7d82b000-7fed7d82c000 rw-p 00010000 08:02 275406                     /usr/lib64/lxcfs/liblxcfs.so
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7d82c000-7fed7d834000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7d834000-7fed7d9f7000 r-xp 00000000 08:02 264428                     /usr/lib64/libc-2.17.so
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7d9f7000-7fed7dbf6000 ---p 001c3000 08:02 264428                     /usr/lib64/libc-2.17.so
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7dbf6000-7fed7dbfa000 r--p 001c2000 08:02 264428                     /usr/lib64/libc-2.17.so
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7dbfa000-7fed7dbfc000 rw-p 001c6000 08:02 264428                     /usr/lib64/libc-2.17.so
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7dbfc000-7fed7dc01000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7dc01000-7fed7dc18000 r-xp 00000000 08:02 264454                     /usr/lib64/libpthread-2.17.so
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7dc18000-7fed7de17000 ---p 00017000 08:02 264454                     /usr/lib64/libpthread-2.17.so
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7de17000-7fed7de18000 r--p 00016000 08:02 264454                     /usr/lib64/libpthread-2.17.so
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7de18000-7fed7de19000 rw-p 00017000 08:02 264454                     /usr/lib64/libpthread-2.17.so
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7de19000-7fed7de1d000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7de1d000-7fed7de1f000 r-xp 00000000 08:02 264434                     /usr/lib64/libdl-2.17.so
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7de1f000-7fed7e01f000 ---p 00002000 08:02 264434                     /usr/lib64/libdl-2.17.so
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7e01f000-7fed7e020000 r--p 00002000 08:02 264434                     /usr/lib64/libdl-2.17.so
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7e020000-7fed7e021000 rw-p 00003000 08:02 264434                     /usr/lib64/libdl-2.17.so
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7e021000-7fed7e04c000 r-xp 00000000 08:02 275263                     /usr/lib64/libfuse.so.2.9.2
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7e04c000-7fed7e24c000 ---p 0002b000 08:02 275263                     /usr/lib64/libfuse.so.2.9.2
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7e24c000-7fed7e25e000 r--p 0002b000 08:02 275263                     /usr/lib64/libfuse.so.2.9.2
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7e25e000-7fed7e25f000 rw-p 0003d000 08:02 275263                     /usr/lib64/libfuse.so.2.9.2
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7e25f000-7fed7e281000 r-xp 00000000 08:02 264421                     /usr/lib64/ld-2.17.so
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7e2f9000-7fed7e2fa000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7e31c000-7fed7e31f000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7e385000-7fed7e388000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7e3e7000-7fed7e3ee000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7e40f000-7fed7e410000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7e44e000-7fed7e47b000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7e47b000-7fed7e480000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7e480000-7fed7e481000 r--p 00021000 08:02 264421                     /usr/lib64/ld-2.17.so
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7e481000-7fed7e482000 rw-p 00022000 08:02 264421                     /usr/lib64/ld-2.17.so
Aug 17 14:00:09 rh-vz01 lxcfs: 7fed7e482000-7fed7e483000 rw-p 00000000 00:00 0
Aug 17 14:00:09 rh-vz01 lxcfs: 7ffc4558e000-7ffc455af000 rw-p 00000000 00:00 0                          [stack]
Aug 17 14:00:09 rh-vz01 lxcfs: 7ffc455d3000-7ffc455d5000 r-xp 00000000 00:00 0                          [vdso]
Aug 17 14:00:09 rh-vz01 lxcfs: ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]
Aug 17 14:00:09 rh-vz01 systemd: Unit lxcfs.service entered failed state.
Aug 17 14:00:09 rh-vz01 systemd: lxcfs.service failed.
Aug 17 14:00:09 rh-vz01 systemd: lxcfs.service holdoff time over, scheduling restart.
Aug 17 14:00:09 rh-vz01 lxcfs: mount namespace: 5
Aug 17 14:00:09 rh-vz01 lxcfs: hierarchies:
Aug 17 14:00:09 rh-vz01 lxcfs: 0: fd:   6: cpuset
Aug 17 14:00:09 rh-vz01 lxcfs: 1: fd:   7: perf_event
Aug 17 14:00:09 rh-vz01 lxcfs: 2: fd:   8: pids
Aug 17 14:00:09 rh-vz01 lxcfs: 3: fd:   9: cpuacct,cpu
Aug 17 14:00:09 rh-vz01 lxcfs: 4: fd:  10: freezer
Aug 17 14:00:09 rh-vz01 lxcfs: 5: fd:  11: hugetlb
Aug 17 14:00:09 rh-vz01 lxcfs: 6: fd:  12: blkio
Aug 17 14:00:09 rh-vz01 lxcfs: 7: fd:  13: memory
Aug 17 14:00:09 rh-vz01 lxcfs: 8: fd:  14: net_prio,net_cls
Aug 17 14:00:09 rh-vz01 lxcfs: 9: fd:  15: devices
Aug 17 14:00:09 rh-vz01 lxcfs: 10: fd:  16: name=systemd
Aug 17 14:00:11 rh-vz01 lxcfs: bindings.c: 2420: pid_from_ns: Timeout reading from parent.
```

### PR DESCRIPTION
**Pull request by brauner**
Closes #252.

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

### CODE REVIEW

**Comment by brauner** [comment]
Potentially closes the crash that has been reported.

**Comment by tomponline** [comment]
cheers @brauner I will retest ASAP

File: bindings.c [hunk 1]
```diff
@@ -2054,6 +2054,7 @@ static void do_release_file_info(struct fuse_file_info *fi)
 	free(f->buf);
 	f->buf = NULL;
 	free(f);
+	f = NULL;
```
[the remark this answers is not in the document]
Comment (tomponline) [reply bindings.c]: thanks @hallyn is there something I can retest?

[the remark this answers is not in the document]
Comment (hallyn) [reply bindings.c]: No, fi->fh = 0 happens higher up, so that's not it.

**Pull request merged**